jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -1043,7 +1043,6 @@ public static String getProperty(String prop, String defaultValue) {
 
 	if (!propertiesInitialized
 			&& !prop.equals("com.ibm.IgnoreMalformedInput") //$NON-NLS-1$
-			&& !prop.equals("file.encoding.pkg") //$NON-NLS-1$
```

Comment:
Should line 748 be removed as well?
```java
	initializedProperties.put("file.encoding.pkg", "sun.io"); //$NON-NLS-1$ //$NON-NLS-2$
```

Reply:
No, this matches the OpenJDK behavior.

Reply:
So the "during boot" is the important qualifier?

Reply:
Yes, this code is only used when `!propertiesInitialized`, which is during bootstrap. Looking at the OpenJDK 8 code, there is no code that looks up "file.encoding.pkg".